lib/Reader.php

```diff
@@ -298,14 +298,14 @@ public function getDeserializerForElementName(string $name): callable
         }
 
         $deserializer = $this->elementMap[$name];
-        if (is_subclass_of($deserializer, 'Sabre\\Xml\\XmlDeserializable')) {
-            return [$deserializer, 'xmlDeserialize'];
-        }
-
         if (is_callable($deserializer)) {
             return $deserializer;
         }
 
+        if (is_subclass_of($deserializer, 'Sabre\\Xml\\XmlDeserializable')) {
```

Comment:
`is_subclass_of` takes a class name or object in parameter 1.
But `$deserializer` can be a callable function.
phpstan is happy if the check for `is_callable` comes first.